### ISSUE TITLE
Magisk 20.4 fails to boot after update to Android 10 (same with 20.3, 19.4, 19.1)

### ISSUE THREAD

**Issue by SoftwareSchlosser** [topjohnwu/Magisk#2695]
I have a similar issue as seen in #2049 with my Xiaomi Mi A2 (non-lite) which had stock Pie with Magisk 19.x (can't remember) running.
I wanted to apply the latest OTA update to Android 10 (security update from March 1st).
After uninstalling Magisk and applying the OTA, the phone boots the stock bootloader as expected.
Now I fastbooted into TWRP and extracted and copied both boot images.
I'm on slot b - so back in Android I patched the corresponding boot image via Magisk Manager and tried to fastboot it. The colored line starts to move which means the boot process itself works but it runs like that forever. I also tried to fastboot the extracted unpatched image which also works so there must be something wrong with the patched image.

[stock boot image](https://github.com/SoftwareSchlosser/temp/blob/master/boot_b-2020.04.16.7z)
[patched boot image](https://github.com/SoftwareSchlosser/temp/blob/master/magisk_patched-b-2020.04.16.7z)

Do you have any ideas on that?

_Originally posted by @SoftwareSchlosser in https://github.com/topjohnwu/Magisk/issues/2049#issuecomment-614779955_

It seems like TWRP can't handle the encryption anymore on Android 10. Maybe that is related?

**Comment by osm0sis** [comment]
You should figure out exactly what was the last version that worked.

**Comment by SoftwareSchlosser** [comment]
It was 20.3 but that was on Pie like I said and even the same version doesn't work since the update to Android 10.

**Comment by 666zam** [comment]
I can't get magisk installed on los 17.1 , Razer phone. Ends in a boot loop.

**Comment by osm0sis** [comment]
Can you try on the latest Canary. The whole init logic was rewritten, so it might fix this.

**Comment by SoftwareSchlosser** [comment]
So Magisk Manager (765d5d97 (289) (9) offers me Magisk 20.4.05 (8ca5a048) after switching to Canary channel. If that's the right version then it unfortunately also doesn't work.

**Comment by osm0sis** [comment]
None of that makes sense. Manager is 283, and Magisk is 20405 right now.

**Comment by SoftwareSchlosser** [comment]
Ok manager is 283 - that was a typo. For magisk I added two dots between the number so if you remove them 20.4.05

**Comment by osm0sis** [comment]
Nope. It isn't represented by anything, it's just the version code. Anyway, guess it didn't work?

**Comment by SoftwareSchlosser** [comment]
So "20.4.05" is not the version string for version code 20405? What a coincidence then.
Anyways it unfortunately doesn't work.

**Comment by osm0sis** [comment]
Correct. There is no version number for Canary builds anymore.

**Comment by mhier** [comment]
I have similar issues. I have a Sony Xperia XA2 with LineageOS. After upgrading to LineageOS 17.1 (actually fresh install), I run into boot loops after installing Magisk. What helps is wiping Dalivk Cache, but only for one single time. So I have to do this now before every reboot. (Luckily TWRP works for me even with encryption...)

I have checked this with the latest Canary: 2468f5a6 (20406) for Magisk and 765d5d97 (283) for the Manager.

I cannot provide a logcat, since this seems to happen before adb gets initialised. I only see it "- waiting for device -"...

@SoftwareSchlosser can you try if wiping dalvik cache helps?

**Comment by SoftwareSchlosser** [comment]
> @SoftwareSchlosser can you try if wiping dalvik cache helps?

Just tried that (also with 2468f5a) but it didn't help :( A log would be nice but I can't get one via adb either.

**Comment by victornworgu** [comment]
i hate that this is an official forum for issues relating to magisk but all I get is closing my threads without offering solutions. i still have bootloops on infinix x604 (android 10) after flashing patched boot with both the latest stable and canary versions. i hope someone could at least point me in the right direction

**Comment by 666zam** [comment]
Same here. Welcome to the club!

Den tors 30 apr. 2020 22:12victornworgu <notifications@github.com> skrev:

**Comment by NihilistBrew** [comment]
This is so goddamn annoying. Bootloop on android 10, LineageOS 17.1 having tried both latest Canary and the normal 20.4 release.

**Comment by rhalfkann** [comment]
Same problem here, fresh installed LOS 17.1 works perfectly until I try to flash Magisk via TWRP. Device gets stuck in a bootloop, unable to get logcat output over ADB.

**Comment by lightvortex** [comment]
+1 bootloops with havoc on mix2, need to go recovery and reboot 2 times for it to boot

**Comment by UltraBlackLinux** [comment]
for me infinite reload on los 17.1 on chery too. Entering the recovery and hitting regoot system now from THERE lets me boot to my sys

**Comment by ChachyDev** [comment]
The solution doesn't seem to work for me, using Android Q SM-A600FN

**Comment by SoftwareSchlosser** [comment]
Tried 99ef206 (20412) but it also didn't work. Why has this been closed? Was it supposed to be fixed?

**Comment by parkerlreed** [comment]
Bootloop on latest Q on Pixel 3 XL. logcat http://ix.io/2mMs

**Comment by mhier** [comment]
Now I realise that I actually don't have to wipe the dalvik cache, just entering twrp is sufficient.

@topjohnwu This issue is not fixed, please reopen or point us to another issue where the bug is tracked. Cheers!

**Comment by osm0sis** [comment]
Possibly #2806 is a currently open similar issue. I won't reopen this one without @topjohnwu's input however.

**Comment by lightvortex** [comment]
i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc 
add this in vendor/etc/hw/init.qcom.rc
`on fs
    wait /dev/block/platform/soc/${ro.boot.bootdevice}
    symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice

    mount_all /vendor/etc/fstab.qcom`

https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e

**Comment by parkerlreed** [comment]
Yeah won't work for those of us that can't run TWRP.

**Comment by parkerlreed** [comment]
@osm0sis I mean this issue is older than the one you linked to. Why would that take precedence?

**Comment by osm0sis** [comment]
I told you, John closed it so I won't reopen it without his acknowledgement. Not sure how else to spell that out.

**Comment by SoftwareSchlosser** [comment]
@topjohnwu this is still a problem. What can we do to help solving it?

**Comment by heldzz** [comment]
> 
> 
> i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc
> add this in vendor/etc/hw/init.qcom.rc
> ` on fs
> wait /dev/block/platform/soc/${ro.boot.bootdevice}
> symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice
> 
> ```
> mount_all /vendor/etc/fstab.qcom`
> ```
> 
> [lightvortex/device_xiaomi_msm8998-common@e57bd5b](https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e)

solved it for me. I'm on a Sony XA2 with lineageOS 17.1. The entry mentioned there was already part of vendor/etc/hw/init.qcom.rc, but with 

mount_all /vendor/etc/fstab.qcom --early

Just deleted --early und rebooted like a charm. Will this be replaced after an OTA update? I'll check that on the build tomorrow...

**Comment by SoftwareSchlosser** [comment]
I'm not on LOS but on Xiaomi stock firmware. How could I benefit from this kind of fix?

**Comment by osm0sis** [comment]
I've personally asked @topjohnwu to review this again, especially since the nature of the regression might be revealed by the workaround you guys have found. I'll open this again in the meantime.

**Comment by heldzz** [comment]
The linux under the hood can (should) be the same. Just search for the file and edit it, if it's available...

**Comment by osm0sis** [comment]
If someone could please try old Magisk canaries and tell us when the issue first happened that would also be helpful, since I don't think we ever figured that out here.

https://github.com/topjohnwu/magisk_files/commits/canary

**Comment by SoftwareSchlosser** [comment]
> The linux under the hood can (should) be the same. Just search for the file and edit it, if it's available...

If that means I would have to edit a file on the system partition of my phone then that's absolutely no option because unfortunately Xiaomis firmware updater refuses to do the update if the system partition has been altered in any way. I once did that and had to wipe and re-flash everything (incl. data) which was a real pain.

**Comment by lightvortex** [comment]
> > i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc
> > add this in vendor/etc/hw/init.qcom.rc
> > ` on fs
> > wait /dev/block/platform/soc/${ro.boot.bootdevice}
> > symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice
> > ```
> > mount_all /vendor/etc/fstab.qcom`
> > ```
> > 
> > 
> > [lightvortex/device_xiaomi_msm8998-common@e57bd5b](https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e)
> 
> solved it for me. I'm on a Sony XA2 with lineageOS 17.1. The entry mentioned there was already part of vendor/etc/hw/init.qcom.rc, but with
> 
> mount_all /vendor/etc/fstab.qcom --early
> 
> Just deleted --early und rebooted like a charm. Will this be replaced after an OTA update? I'll check that on the build tomorrow...

This is the proper fix if you need it 
https://review.lineageos.org/c/LineageOS/android_device_xiaomi_msm8998-common/+/276735

**Comment by parkerlreed** [comment]
For some people this isn't an option. For example the Pixels can't use TWRP and thus cannot edit the filesystem outside of the OS.

**Comment by osm0sis** [comment]
So now we'll need to accommodate that Lineage patch too, most likely.🤦‍♂️

**Comment by parkerlreed** [comment]
I am using a Pixel... Granted I havent tried in a little bit. The first comment from me here is it bootlooping with a log on the Pixel 3 XL

**Comment by parkerlreed** [comment]
Ok whatever it was on my device, it is working now. Thanks for the help.

**Comment by SeanChengN** [comment]
> > > i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc
> > > add this in vendor/etc/hw/init.qcom.rc
> > > ` on fs
> > > wait /dev/block/platform/soc/${ro.boot.bootdevice}
> > > symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice
> > > ```
> > > mount_all /vendor/etc/fstab.qcom`
> > > ```
> > > 
> > > 
> > > [lightvortex/device_xiaomi_msm8998-common@e57bd5b](https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e)
> > 
> > 
> > solved it for me. I'm on a Sony XA2 with lineageOS 17.1. The entry mentioned there was already part of vendor/etc/hw/init.qcom.rc, but with
> > mount_all /vendor/etc/fstab.qcom --early
> > Just deleted --early und rebooted like a charm. Will this be replaced after an OTA update? I'll check that on the build tomorrow...
> 
> This is the proper fix if you need it
> https://review.lineageos.org/c/LineageOS/android_device_xiaomi_msm8998-common/+/276735

It doesn't seem to be merged, but why?

**Comment by Tooniis** [comment]
I had this issue after upgrading my Mi Note 2 (scorpio) to LineageOS 17.1 too.

> i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc
> add this in vendor/etc/hw/init.qcom.rc
> ` on fs
> wait /dev/block/platform/soc/${ro.boot.bootdevice}
> symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice
> 
> ```
> mount_all /vendor/etc/fstab.qcom`
> ```
> 
> [lightvortex/device_xiaomi_msm8998-common@e57bd5b](https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e)

Is this specific to MSM8998 devices? I couldn't find that file on mine which has a MSM8996 Pro

**Comment by lightvortex** [comment]
> I had this issue after upgrading my Mi Note 2 (scorpio) to LineageOS 17.1 too.
> 
> > i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc
> > add this in vendor/etc/hw/init.qcom.rc
> > ` on fs
> > wait /dev/block/platform/soc/${ro.boot.bootdevice}
> > symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice
> > ```
> > mount_all /vendor/etc/fstab.qcom`
> > ```
> > 
> > 
> > [lightvortex/device_xiaomi_msm8998-common@e57bd5b](https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e)
> 
> Is this specific to MSM8998 devices? I couldn't find that file on mine which has a MSM8996 Pro

Yup it works on msm8996 too tested on my Nubia z11

**Comment by Tooniis** [comment]
> Check system/vendor/etc/init/hw

`system` only has `usr` and `bin` in it.
I ran `grep -r init.qcom.rc` for a while but it found nothing

**Comment by Tooniis** [comment]
> Check system/vendor/etc/init/hw

`system` only has `usr` and `bin` in it.
I ran `grep -r init.rc /` for a while but it found nothing

**Comment by Tooniis** [comment]
> > I had this issue after upgrading my Mi Note 2 (scorpio) to LineageOS 17.1 too.
> > > i found the fix apparently magisk fails to mount fstab during boot so we can bypass that by adding mount in init.qcom.rc
> > > add this in vendor/etc/hw/init.qcom.rc
> > > ` on fs
> > > wait /dev/block/platform/soc/${ro.boot.bootdevice}
> > > symlink /dev/block/platform/soc/${ro.boot.bootdevice} /dev/block/bootdevice
> > > ```
> > > mount_all /vendor/etc/fstab.qcom`
> > > ```
> > > 
> > > 
> > > [lightvortex/device_xiaomi_msm8998-common@e57bd5b](https://github.com/lightvortex/device_xiaomi_msm8998-common/commit/e57bd5bedd2b7eeca4c323003e1b21318c42108e)
> > 
> > 
> > Is this specific to MSM8998 devices? I couldn't find that file on mine which has a MSM8996 Pro
> 
> Yup it works on msm8996 too tested on my Nubia z11
> 
> Check system/vendor/etc/init/hw

Turns out I just didn't have `vendor` mounted.

Apparently those lines are already there, and it booted now?
I have no idea what changed to make this happen.

**Comment by cbaptista** [comment]
Hi all,
I have the same problem, the boot goes for ever, with oneplus 2.
None of the above workarounds worked.

Thanks
CB

**Comment by Fox2Code** [comment]
@cbaptista Can you make your phone boot loop and provide `last_kmsg` via TWRP 
(`last_kmsg` is the kernel logs of the previous boot so you need to get them via your recovery)

**Comment by cbaptista** [comment]
@Fox2Code I can, but only after next thursday . I will post them here then.

CB

**Comment by cbaptista** [comment]
@Fox2Code there is no /proc/last_kmsg.

**Comment by fgl27** [comment]
The [device](https://github.com/fgl27/device_motorola_quark#device-configuration-for-moto-maxx-quark) work well on android 9 and older, but after android 10 it never worked.

Using latest version from release **Magisk v20.4** the magisk manager doesn't recognizes magisk su device boot OK.

Using latest version **20418** the device doesn't finises the boot process.

It also looses adb root, so logcat is limited, I took some logs from **20418**
[logcat.txt](https://github.com/topjohnwu/Magisk/files/4861473/logcat.txt)

Can't run `dmesg` because no SU, but via TWRP I can `cat console-ramoops` AKA `dmesg`
[console-ramoops.txt](https://github.com/topjohnwu/Magisk/files/4861474/console-ramoops.txt)

There is some odd `Unmounting` spaming on dmesg, data and cache are formatted as f2fs don't know if any relation have too many thing on the device data do change it to ext4, but will try it when I have some free time to backup it all.

Anyone has some experience with this issue?

**Comment by singh-manbir** [comment]
> I have a similar issue as seen in #2049 with my Xiaomi Mi A2 (non-lite) which had stock Pie with Magisk 19.x (can't remember) running.
> I wanted to apply the latest OTA update to Android 10 (security update from March 1st).
> After uninstalling Magisk and applying the OTA, the phone boots the stock bootloader as expected.
> Now I fastbooted into TWRP and extracted and copied both boot images.
> I'm on slot b - so back in Android I patched the corresponding boot image via Magisk Manager and tried to fastboot it. The colored line starts to move which means the boot process itself works but it runs like that forever. I also tried to fastboot the extracted unpatched image which also works so there must be something wrong with the patched image.
> 
> [stock boot image](https://github.com/SoftwareSchlosser/temp/blob/master/boot_b-2020.04.16.7z)
> [patched boot image](https://github.com/SoftwareSchlosser/temp/blob/master/magisk_patched-b-2020.04.16.7z)
> 
> Do you have any ideas on that?
> 
> _Originally posted by @SoftwareSchlosser in [#2049 (comment)](https://github.com/topjohnwu/Magisk/issues/2049#issuecomment-614779955)_
> 
> It seems like TWRP can't handle the encryption anymore on Android 10. Maybe that is related?

I had the same issue it is solved now, i tried all the version 13, 15, 17, 19 ,20.4 and canary but the issue was solved when I used the version 20.0 of magisk. 

If your phone is not booting after magisk flash or it is in bootloop, the only solution that I have found is that magisk 20.0 version works fine. Hope you find this helpful.

**Comment by SoftwareSchlosser** [comment]
> I had the same issue it is solved now, i tried all the version 13, 15, 17, 19 ,20.4 and canary but the issue was solved when I used the version 20.0 of magisk.
> 
> If your phone is not booting after magisk flash or it is in bootloop, the only solution that I have found is that magisk 20.0 version works fine. Hope you find this helpful.

I tried that version and even downgraded Magisk Manager to the version shown on your screenshot but it doesn't even display any boot- / splash logo. Just a black screen after the `fastboot boot ...` command. Also cleared the cache partitions in TWRP but that didn't help too. I can boot the unpatched boot image so there must be something wrong with the patched file.

**Comment by SoftwareSchlosser** [comment]
Hello again, this is the last line from my failing boot log:
`mdss_fb_do_ioctl: unsupported ioctl (5401)`
Does this help?

**Comment by nazar-pc** [comment]
Initially reported by another user here (and I left a few comments): https://gitlab.com/LineageOS/issues/android/-/issues/2300

Essentially as soon as Magisk is installed on OnePlus 7 Pro on either stock OxygenOS or LineageOS 17.1 it either causes bootloop (most of the time was fixed by "Fix Recovery Bootloop" in TWRP for some reason) and WiFi/voice calls not working (LTE data still works). I had 20.4 running just fine until last update though.
It happens in such a way that uninstallation or re-flashing OS doesn't help, only complete reflashing of a bunch of partitions to completely stock phone state.

There have been no comments here for a while, is this fixed in nightly builds?

**Comment by osm0sis** [comment]
I always run Magisk Canary and Lineage 17.1 with TWRP installed on my OP7Pro without issue.

**Comment by nazar-pc** [comment]
Probably a stupid question, but readme mentions "Canary Magisk Manager", where can I get it?

**Comment by osm0sis** [comment]
Tap the download count badge for the Canary Manager, it'll grab everything else for you

**Comment by nazar-pc** [comment]
Update to Canary didn't fix WiFi. There was sonething in `/data` left by Magisk 20.4 that was destroying WiFi on all slots right after first boot with `/data` restored from backup.

With everything wiped once again, fresh LineageOS 2020-08-09 installed on top + opengapps + Canary Magisk I can finally both have Magisk and working WiFi again. Yay!

Please release new stable version then, 20.4 clearly breaks Android 10 devices and Canary (in my case at least) does not.

**Comment by osm0sis** [comment]
Magisk doesn't touch anything except boot.img and put its own stuff in /data/adb (and /system/addon.d on custom ROMs). Any lasting effect basically isn't Magisk's fault.

But anyway, obviously another stable will happen, now stop polluting this thread

**Comment by osm0sis** [comment]
Might be worth trying latest Canary again, SAR detection and init logic was rewritten. 👍

**Comment by fgl27** [comment]
> The [device](https://github.com/fgl27/device_motorola_quark#device-configuration-for-moto-maxx-quark) work well on android 9 and older, but after android 10 it never worked.
> 
> Using latest version from release **Magisk v20.4** the magisk manager doesn't recognizes magisk su device boot OK.
> 
> Using latest version **20418** the device doesn't finises the boot process.
> 
> It also looses adb root, so logcat is limited, I took some logs from **20418**
> [logcat.txt](https://github.com/topjohnwu/Magisk/files/4861473/logcat.txt)
> 
> Can't run `dmesg` because no SU, but via TWRP I can `cat /sys/fs/pstore/console-ramoops` AKA `dmesg`
> [console-ramoops.txt](https://github.com/topjohnwu/Magisk/files/4861474/console-ramoops.txt)
> 
> There is some odd `Unmounting` spamming on dmesg of data and cache, that doesn't happen on normal boot.
> 
> Anyone has some experience with this issue?

I fix my problem...

Regarding the release version 20.4 (20400), it doesn't work because it can't patch the dtb this devices was using a separated zImage and dtb , to fix this issue I add the kernel defconfig flags

```
CONFIG_BUILD_ARM_APPENDED_DTB_IMAGE=y
CONFIG_BUILD_ARM_APPENDED_DTB_IMAGE_NAMES=""
```

Removed on dt BoardConfig.mk 

`BOARD_KERNEL_SEPARATED_DT := true
`

Technically the canary version 20420 and up doesn't have this issue regarding the dtb (I assume the dtb issue was fixed on a version prior to 20420 that was the one I tested after 20400), but it will not boot the device just stay on the `bootanimation` screen for ever...

So to fix that issue I changed the data partition to `latemount`

https://github.com/fgl27/device_motorola_quark/commit/bec4069d85543b261e558a4200c60d91c62521b1

I did other things on the above, but what is need there is to add `latemount` to the data partition in fstab and change the mount process of fstab
```

on fs
    mount_all /vendor/etc/fstab.qcom --early

on late-fs
    mount_all /vendor/etc/fstab.qcom --late
```

So for some reason that I don't know magisk need a "delay" at init, as it is inside the data partition late mouting it resolves the issue for this device,  but I do believed that with this information some improves on magiks can be made to address the problem.

regarding 20423, it doesn't work, the last that work for me is 20422... using 20423 the device boot but magisk manager just freezes and no app can request SU. I know that is a wipe version for Android 11 but extra info always help...

In doubt about my process let me know.

**Comment by osm0sis** [comment]
Try 20424, there was an selinux issue

**Comment by AnErW** [comment]
> Might be worth trying latest Canary again, SAR detection and init logic was rewritten. 👍

Thanks, Try It later.

**Comment by UltraBlackLinux** [comment]
Canary Fixed nothing for me, sadly.
